src/plugins/intel_cpu/src/async_infer_request.cpp

```diff
@@ -29,3 +30,11 @@ void ov::intel_cpu::AsyncInferRequest::setSubInferRequest(
     const std::vector<std::shared_ptr<IAsyncInferRequest>>& requests) {
     m_sub_infer_requests = requests;
 }
+
+void ov::intel_cpu::AsyncInferRequest::infer() {
+    if (m_is_single_thread) {
```

Comment:
Consider to store member function pointer to `infer()` instead bool flag and set during construction (to this class or base) 
Here just execute it without check.